test/slax/github_test.exs

```diff
@@ -542,4 +542,66 @@ defmodule Slax.Github.Test do
       assert Github.add_team_to_repo(params) == {:error, "Invalid"}
     end
   end
+
+  def load_issue_setup(context) do
+    project = insert(:project)
+    insert(:project_repo, project: project, token: "success", repo_name: "success")
+    insert(:project_repo, project: project, token: "failure", repo_name: "failure")
+    insert(:project_repo, project: project, token: nil, repo_name: "nil")
+
+    params = %{
+      repo_and_issue_success: "success/1",
+      repo_and_issue_failure: "failure/1",
+      repo_and_issue_nil: "nil/1",
+      repo_and_issue_na: "na/1"
+    }
+
+    {:ok, context |> Map.put(:params, params)}
+  end
+
+  describe "load_issue/1" do
+    setup [:load_issue_setup]
+
+    test "success", %{params: %{repo_and_issue_success: repo_and_issue}} do
+      expect(Slax.Tentacat.IssuesMock, :find, fn _, _, _, _ ->
+        {200, %{issue: "success"}, %{}}
+      end)
+
+      assert {:ok, %{issue: "success"}, ""} == Github.load_issue(repo_and_issue)
+    end
+
+    test "failure: access token invalid", %{params: %{repo_and_issue_failure: repo_and_issue}} do
+      expect(Slax.Tentacat.IssuesMock, :find, fn _, _, _, _ ->
+        {401, %{"message" => "Bad credentials"}, %{}}
+      end)
+
+      assert {:error, "Access token invalid for #{repo_and_issue}"} ==
+               Github.load_issue(repo_and_issue)
+    end
+
+    test "failure: no access token", %{params: %{repo_and_issue_nil: repo_and_issue}} do
+      expect(Slax.Tentacat.IssuesMock, :find, fn _, _, _, _ ->
+        {200, %{issue: "success"}, %{}}
+      end)
+
+      assert {:ok, %{issue: "success"}, "(Please setup a fine grained access token with /token)"} ==
+               Github.load_issue(repo_and_issue)
+
+      # NOTE: remove expect and add back these asserts once the guard is removed and the app no longer uses the generic access token
```

Comment:
do we have the details of this work / plan written up in an issue yet?

Reply:
No I'll setup an issue for it